test_stokes_wrapper fails on arm64

We have a [test failure](https://ci.debian.net/data/autopkgtest/testing/arm64/g/gwcs/3916974/log.gz), only on arm64, which I do not completely understand:
```
_____________________________ test_stokes_wrapper ______________________________

gwcs_stokes_lookup = <WCS(output_frame=StokesFrame, input_frame=detector, forward_transform=Model: Tabular1D
N_inputs: 1
N_outputs: 1
Param...ntity [0., 1., 2., 3.] pix>,)
  lookup_table: [0. 1. 2. 3.]
  method: nearest
  fill_value: nan
  bounds_error: False)>

    def test_stokes_wrapper(gwcs_stokes_lookup):
        pytest.importorskip("astropy", minversion="4.0dev0")
[…]
        pixel_input = [-1, 4]
    
>       out = hlvl.pixel_to_world(pixel_input*u.pix)

/usr/lib/python3/dist-packages/gwcs/tests/test_api.py:287: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/usr/lib/python3/dist-packages/astropy/wcs/wcsapi/high_level_api.py:252: in pixel_to_world
    result.append(klass_gen(*args[key], *ar, **kwargs[key], **kw))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

cls = <class 'gwcs.coordinate_frames.StokesProfile'>, indexes = array([0, 0])

    @classmethod
    def from_index(cls, indexes):
        """[…]"""
        nans = np.isnan(indexes)
        indexes = np.asanyarray(indexes, dtype=int)
        out = np.empty_like(indexes, dtype=object)
    
        out[nans] = np.nan
    
        for profile, index in cls.profiles.items():
            out[indexes == index] = profile
    
        if out.size == 1 and not nans:
            return StokesProfile(out.item())
        elif nans.all():
>           return np.array(out, dtype=float)
E           ValueError: could not convert string to float: 'I'

/usr/lib/python3/dist-packages/gwcs/coordinate_frames.py:640: ValueError
```

When I try to reproduce the `from_index` class method: https://github.com/spacetelescope/gwcs/blob/0555a049eb40e6a8f608a347534fe89e4c1f61f8/gwcs/coordinate_frames.py#L599-L615

This basically fills the `out` array with the keys of this map, so specifically with strings like `'I'`: https://github.com/spacetelescope/gwcs/blob/0555a049eb40e6a8f608a347534fe89e4c1f61f8/gwcs/coordinate_frames.py#L634-L635

which then can't converted to float. Buy why does this work on amd64? What do I miss here?
Astropy version is 4.0.

--- a/gwcs/coordinate_frames.py
+++ b/gwcs/coordinate_frames.py
@@ -629,10 +629,10 @@ class StokesProfile(str):
         indexes = np.asanyarray(indexes, dtype=int)
         out = np.empty_like(indexes, dtype=object)
 
-        out[nans] = np.nan
-
         for profile, index in cls.profiles.items():
             out[indexes == index] = profile
+
+        out[nans] = np.nan
 
         if out.size == 1 and not nans:
             return StokesProfile(out.item())

